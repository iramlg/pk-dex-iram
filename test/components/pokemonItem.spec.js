const React = require('react');
const PokemonItem = require('../../src/components/pokemonItem');
const renderer = require('react-test-renderer');
const { shallow } = require('enzyme');

describe('<PokemonItem />', () => {
  describe('render', () => {
    it('test rendering', () => {
      const component = renderer.create(<PokemonItem getPokemon={ jest.fn() } pokemon={ {name: 'name', url: 'url'} } />).toJSON();
      expect(component).toMatchSnapshot();
    });

    it('test render instance', () => {
      const handleGetPokemon = jest.fn();
      const localInstance = shallow(<PokemonItem getPokemon={ handleGetPokemon } pokemon={ {name: 'name', url: 'url'} } />);
      localInstance.simulate('click');
      expect(handleGetPokemon).toHaveBeenCalled();
    });
  });

  describe('Props', () => {
    it('types', () => {
      expect(Object.keys(PokemonItem.propTypes).length).toBe(2);

      ['pokemon', 'getPokemon'].forEach(prop => {
        expect(PokemonItem.propTypes[prop]).toBeTruthy();
      });

    });
  });
});
