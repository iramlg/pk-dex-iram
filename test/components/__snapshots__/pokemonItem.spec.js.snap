// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonItem /> render test rendering 1`] = `
<div
  className="pokemon-item"
  onClick={[Function]}
>
  <span>
    NAME
  </span>
</div>
`;
